targets/adcSample/main.cpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * Tests the values produced by the ADC along with a rolling average voltage
+ */
+#include <EVT/io/ADC.hpp>
+#include <EVT/io/GPIO.hpp>
+#include <EVT/io/manager.hpp>
+#include <EVT/utils/time.hpp>
+
+namespace IO = EVT::core::IO;
+
+namespace time = EVT::core::time;
+
+uint32_t calcAvg(uint32_t lastTen[]) {
+    uint32_t avg = 0;
+    for (int i = 0; i < 10; i++) {
+        avg += lastTen[i];
+    }
+    return avg / 10;
+}
+
+void switchMux(IO::GPIO& muxs3, IO::GPIO& muxs2, IO::GPIO& muxs1, uint8_t s3, uint8_t s2, uint8_t s1) {
+    s3 == 0 ? muxs3.writePin(IO::GPIO::State::LOW) : muxs3.writePin(IO::GPIO::State::HIGH);
+    s2 == 0 ? muxs2.writePin(IO::GPIO::State::LOW) : muxs2.writePin(IO::GPIO::State::HIGH);
+    s1 == 0 ? muxs1.writePin(IO::GPIO::State::LOW) : muxs1.writePin(IO::GPIO::State::HIGH);
+}
+
+int main() {
+    IO::init();
+
+    IO::ADC& adc = IO::getADC<IO::Pin::PA_4>();
+    IO::UART& uart = IO::getUART<IO::Pin::PB_10, IO::Pin::PB_11>(115200);
+
+    IO::GPIO& muxs1 = IO::getGPIO<IO::Pin::PB_2>();
+    IO::GPIO& muxs2 = IO::getGPIO<IO::Pin::PB_8>();
+    IO::GPIO& muxs3 = IO::getGPIO<IO::Pin::PB_9>();
+
+    uint32_t rollAvg = 0;
+    uint32_t lastTen[] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+    unsigned int counter = 0;
+    unsigned int looper = 1;
+    uint32_t currV1 = -1;
```

Comment:
These 4 values would probably make more sense as an array, so you can access them programmatically in the switch statement instead of hardcoding